tests/ui/async-await/dont-ice-for-type-mismatch-in-closure-in-async.stderr

```diff
@@ -6,7 +6,7 @@ LL | |             false
    | |             ^^^^^ expected `()`, found `bool`
 LL | |
 LL | |         }
-   | |_________- expected this to be `()`
+   | |_________- expected this `if` expression to be `()` because it has no `else` arm
```

Comment:
Aren't we pointing at the block? It seems slightly off to call it an `if` expression, although I don't have a good alternative in mind right now.